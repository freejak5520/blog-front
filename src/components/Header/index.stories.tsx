import type { Meta, StoryObj } from "@storybook/react";
import MaxWidthDecorator from "../decorator/MaxWidthDecorator";
import Header from "./index";

const meta = {
  title: "Components/Header",
  component: Header,
  decorators: [(story) => <MaxWidthDecorator>{story()}</MaxWidthDecorator>],
} satisfies Meta<typeof Header>;
export default meta;

type Story = StoryObj<typeof meta>;

export const Default = {
  args: { title: "Home" },
} satisfies Story;
