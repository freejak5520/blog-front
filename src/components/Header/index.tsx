import NavBar from "../NavBar";
import { Separator } from "../ui/separator";
import { H1 } from "../ui/typography";

const Header = ({ title }: { title: string }) => {
  return (
    <div>
      <H1>{title}</H1>
      <NavBar />
      <Separator className="my-10" />
    </div>
  );
};

export default Header;
