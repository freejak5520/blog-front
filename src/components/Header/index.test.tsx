import { render } from "@testing-library/react";
import Header from ".";

describe("Header", () => {
  it("header render test", () => {
    const { container } = render(<Header title={"test Title"}></Header>);

    expect(container).toMatchSnapshot();
  });

  it("header title test", () => {
    const title = "Test title";
    const { container } = render(<Header title={title}></Header>);

    const h1 = container.querySelector("h1");

    expect(h1?.innerHTML).toBe(title);
  });
});
