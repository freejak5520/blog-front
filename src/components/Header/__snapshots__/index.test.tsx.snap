// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header header render test 1`] = `
<div>
  <div>
    <h1
      class="scroll-m-20 text-4xl font-extrabold tracking-tight lg:text-5xl"
    >
      test Title
    </h1>
    <div
      class="flex flex-row justify-end gap-3"
    >
      <a
        class="font-semibold"
        href="/posts"
      >
        Blog
      </a>
      <a
        class="font-semibold"
        href="/about"
      >
        About
      </a>
    </div>
    <div
      class="shrink-0 bg-border h-[1px] w-full my-10"
      data-orientation="horizontal"
      role="none"
    />
  </div>
</div>
`;
